src/extension.ts

```diff
@@ -27,8 +27,13 @@ export function activate(context: ExtensionContext) {
     workspace.onDidCloseTextDocument((document) =>
       validations.delete(document.uri)
     ),
-    workspace.onDidChangeTextDocument((event) =>
-      validations.delete(event.document.uri)
+    workspace.onDidChangeTextDocument(async (event) => {
```

Comment:
This should be according to the settings, such as on save on change etc.

Reply:
I also discovered this issue today and will submit it later